node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -218,81 +201,105 @@ impl PaymentAdjusterReal {
         }
     }
 
-    fn propose_adjustments_recursively<AR, RT>(
-        &mut self,
-        unresolved_accounts: Vec<WeightedPayable>,
-        adjustment_runner: AR,
-    ) -> RT
-    where
-        AR: AdjustmentRunner<ReturnType = RT>,
-    {
-        diagnostics!(
-            "\nUNRESOLVED QUALIFIED ACCOUNTS IN CURRENT ITERATION:",
-            &unresolved_accounts
-        );
+    fn resolve_initial_adjustment_dispatch(
```

Comment:
This function returns `Either`, if you put the exhaustion logic here, you can get rid of the `Either`. Also, you might like to put the exhaustion logic in a different function and then call that function inside this one.